nixos/modules/system/boot/plymouth.nix

```diff
@@ -4,7 +4,10 @@ with lib;
 
 let
 
-  inherit (pkgs) plymouth nixos-icons;
+  inherit (pkgs) nixos-icons;
+  plymouth = pkgs.plymouth.override {
+    systemd = config.boot.initrd.systemd.package;
+  };
```

Comment:
I feel like the plymouth pkg should just always use the stage1 systemd package, no?

Reply:
I believe that's what this line does. Causes it to be built with the systemd that we put into stage 1. It's not specifically the `systemdStage1` package because if users (or e.g. the eventual networkd module) set `config.boot.initrd.systemd.package` to something else, we don't want two systemd paths in initird.